client/src/route/digitaltwins/ExecuteDigitalTwin.tsx

```diff
@@ -0,0 +1,211 @@
+/* eslint-disable no-console */
+
+import React, { useState, useEffect } from 'react';
+import { CardActions, Typography, Button, Dialog, DialogTitle, DialogContent, DialogActions, Snackbar, Alert, AlertColor, CircularProgress } from '@mui/material';
+import { GitlabInstance } from 'util/gitlab';
+import DigitalTwin from 'util/gitlabDigitalTwin';
+import stripAnsi from 'strip-ansi';
+import { getAuthority } from 'util/envUtil';
+import DigitalTwinCard from './DigitalTwinCard';
+
+const formatName = (name: string) =>
+  name
+    .replace(/-/g, ' ')
+    .replace(/^./, (char) => char.toUpperCase());
+
+const ExecuteDigitalTwin: React.FC<{ name: string }> = (props) => {
+  const [gitlabInstance] = useState<GitlabInstance>(new GitlabInstance(sessionStorage.getItem('username') || '', getAuthority(), sessionStorage.getItem('access_token') || ''));
```

Comment:
@nichlaes 
The next 10 lines of react component state seems a bit complex. @VanessaScherma and I are thinking of ways to simplify it without violating react principles. Kindly provide your feedback.

@VanessaScherma we could discuss this after rest of the issues are fixed. We will also get feedback from @nichlaes.

Reply:
@prasadtalasila I think I need a bit more context in regards to what you want to simplify. Is it the number of variables? The initiation of the variables?